lib/Db/TagMapper.php

```diff
@@ -145,20 +145,22 @@ public function getAllTagsForMessages(array $messages): array {
 
 		$qb = $this->db->getQueryBuilder();
 		$idsQuery = $qb->select('mt.*')
+			->addSelect('t.*')
 			->from('mail_message_tags', 'mt')
+			->join('mt', $this->getTableName(), 't', $qb->expr()->eq('mt.tag_id','t.id'))
 			->where(
-				$qb->expr()->in('imap_message_id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_STR_ARRAY))
+				$qb->expr()->in('mt.imap_message_id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_STR_ARRAY))
 			);
 		$idsQuery = $idsQuery->execute();
 		$queryResult = $idsQuery->fetchAll();
```

Comment:
not necessarily related but another microoptimization to get the memory footprint down is to not use `fetchAll` but `fetch` in a loop until you get no more results

Reply:
done